central/declarativeconfig/health/datastore/datastore_impl.go

```diff
@@ -49,3 +50,19 @@ func (ds *datastoreImpl) RemoveDeclarativeConfig(ctx context.Context, id string)
 func (ds *datastoreImpl) GetDeclarativeConfig(ctx context.Context, id string) (*storage.DeclarativeConfigHealth, bool, error) {
 	return ds.store.Get(ctx, id)
 }
+
+func (ds *datastoreImpl) UpdateErrorMessageForDeclarativeConfig(ctx context.Context, id string, errToUpdate error) error {
+	existingHealth, exists, err := ds.GetDeclarativeConfig(ctx, id)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errox.NotFound.Newf("unable to find config health for declarative config %q", id)
```

Comment:
I think we should add resource name to the error by wrapping up the chain - there's not much benefit in specifying non-existent health id if you can't find what related resource is

Reply:
There's no resource name to give "up to the chain" though since in this case we only have the ID here. The errors here are anyways logged, so not sure it really makes a difference tbh.